pkg/frr/frr.go

```diff
@@ -379,24 +372,24 @@ func setUpVrf(vrf *infradb.Vrf) (string, bool) {
 		} else {
 			LbiP = fmt.Sprintf("%+v", vrf.Spec.LoopbackIP.IP)
 		}
-		_, err = Frr.FrrBgpCmd(ctx, fmt.Sprintf("configure terminal\n router bgp %+v vrf %s\n bgp router-id %s\n no bgp ebgp-requires-policy\n no bgp hard-administrative-reset\n no bgp graceful-restart notification\n address-family ipv4 unicast\n redistribute connected\n redistribute static\n exit-address-family\n address-family l2vpn evpn\n advertise ipv4 unicast\n exit-address-family\n exit", localas, path.Base(vrf.Name), LbiP))
-		if err != nil {
+		data, err = Frr.FrrBgpCmd(ctx, fmt.Sprintf("configure terminal\n router bgp %+v vrf %s\n bgp router-id %s\n no bgp ebgp-requires-policy\n no bgp hard-administrative-reset\n no bgp graceful-restart notification\n address-family ipv4 unicast\n redistribute connected\n redistribute static\n exit-address-family\n address-family l2vpn evpn\n advertise ipv4 unicast\n exit-address-family\n exit", localas, path.Base(vrf.Name), LbiP))
+		if err != nil || checkFrrResult(data, false) {
+			log.Printf("FRR: Error Executing config t bgpVrfName router bgp %+v vrf %s bgp_route_id %s no bgp ebgp-requires-policy exit-vrf exit data %v \n", localas, vrf.Name, LbiP, data)
 			return "", false
 		}
 
 		log.Printf("FRR: Executed config t bgpVrfName router bgp %+v vrf %s bgp_route_id %s no bgp ebgp-requires-policy exit-vrf exit\n", localas, vrf.Name, LbiP)
 		// Update the vrf with attributes from FRR
 		cmd := fmt.Sprintf("show bgp l2vpn evpn vni %d json", *vrf.Spec.Vni)
 		cp, err := Frr.FrrBgpCmd(ctx, cmd)
-		if err != nil {
-			log.Printf("error-%v", err)
+		if err != nil || checkFrrResult(cp, true) {
+			log.Printf("FRR Error-show bgp l2vpn evpn vni %v cp %v", err, cp)
 		}
 		hname, _ := os.Hostname()
 		L2vpnCmd := strings.Split(cp, "json")
 		L2vpnCmd = strings.Split(L2vpnCmd[1], hname)
 		cp = L2vpnCmd[0]
-		// fmt.Printf("FRR_L2vpn[0]: %s\n",cp)
-		if len(cp) != 7 {
+		if len(cp) != 7 { // Checking CMD o/p
```

Comment:
magic number. Rely on string length is very unreliable

Reply:
Addressed this and changed to string compare instead of string length

Reply:
is response should be in json format? Should we better try to Unmarshall it then using`"\r\n{}\r\n{\r\n}\r\n"`? If not, maybe we could check if it contains alphanumerical values instead?

Reply:
Changed the comparing the string with alphanumerical.

Reply:
How the original output of FrrBgpCmd look like? I have an impression that you are trying to remove brackets and then add them back

Reply:
Below is the original output of FrrBgpCmd, we trim output for extracting data between the brackets for unmarshal the json o/p. 
	
**acc-1449# show bgp l2vpn evpn vni 3000 json
{
  "vni":3000,
  "type":"L3",
  "inKernel":"True",
  "rd":"10.120.3.1:2",
  "originatorIp":"10.3.3.3",
  "advertiseGatewayMacip":"n/a",
  "advertiseSviMacIp":"n/a",
  "advertisePip":"Enabled",
  "sysIP":"10.3.3.3",
  "sysMac":"00:e8:d4:70:8e:70",
  "rmac":"00:e8:d4:70:8e:70",
  "importRts":[
    "65003:3000"
  ],
  "exportRts":[
    "65003:3000"
  ]
}
acc-1449#**

Reply:
>  we trim output for extracting data between the brackets for unmarshal

Why? We can unmarshal with brackets, no?
https://godbolt.org/z/rxzWGr7q6

Reply:
Changes done